packages/graphile-build-pg/src/plugins/viaTemporaryTable.js

```diff
@@ -83,7 +87,17 @@ export default async function viaTemporaryTable(
          * representation to make it easier to deal with below.
          */
         sql.query`(case when ${sqlResultSourceAlias} is null then null else ${sqlResultSourceAlias} end)`
-      : sql.query`(${sqlResultSourceAlias}.${sqlResultSourceAlias})::${sqlTypeIdentifier}`;
+      : isPgRecord
+        ? sql.query`array[${sql.join(
+            outputArgNames.map(
+              (outputArgName, idx) =>
+                sql.query`${sqlResultSourceAlias}.${sql.identifier(
+                  outputArgName !== "" ? outputArgName : `column${idx + 1}`
```

Comment:
Add a comment here noting that `column1` etc is what PostgreSQL picks currently if there's no name, but that we have no guarantee this will always be the case.

Reply:
Done. eb9382c